scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -475,12 +481,29 @@ object Execution {
    */
 
   private trait ToWrite {
+    def setExtraConfig(conf: Config): Config = Config.empty
+
     def write(config: Config, flowDef: FlowDef, mode: Mode): Unit
   }
+
   private case class SimpleWrite[T](pipe: TypedPipe[T], sink: TypedSink[T]) extends ToWrite {
+
+    var extraConfig = Config.empty
```

Comment:
Generally we avoid mutable stuff like this where at all possible, this config could persist/grow or do different things now depending on re-executions of this execution?